### PR TITLE
tril_indices for numpy frontend

### PR DESCRIPTION
**Pull request by ipriyam26**
closes https://github.com/unifyai/ivy/issues/11016
All tests are passing except the jax ones but they don't seem to be passing for others as well at my end so i guess thats a common issue please review

### CODE REVIEW

**Comment by ipriyam26** [comment]
closes https://github.com/unifyai/ivy/issues/11016

**Comment by juliagsy** [comment]
Hello! Thanks for the great work! Just to double check so that I'm understanding correctly, the tests are not passing for the jax backend for the internal ivy.tril_indices itself, is that correct?

**Comment by ipriyam26** [comment]
@juliagsy  yes other than that its all passing

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py [hunk 1]
```diff
@@ -132,3 +132,31 @@ def test_numpy_take_along_axis(
...
         axis=axis,
     )
+
+
+@handle_frontend_test(
+    fn_tree="numpy.tril_indices",
+    n_rows=helpers.ints(min_value=1, max_value=10),
+    n_cols=helpers.ints(min_value=1, max_value=10),
+    k=st.integers(min_value=-10, max_value=10),
+)
```
Comment (juliagsy) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py:142]: please also add a `test_with_out=st.just(False)` here

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py [hunk 2]
```diff
@@ -132,3 +132,31 @@ def test_numpy_take_along_axis(
...
+    fn_tree,
+    on_device,
+):
+    helpers.test_frontend_function(
+        input_dtypes=["int32"],
+        test_flags=test_flags,
+        frontend=frontend,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        n_rows=n_rows,
```
Comment (juliagsy) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py:159]: the parameters here should then match the function signature/args

**Review by juliagsy — CHANGES_REQUESTED** [review]
Hey! Great work so far! I've just realised some inconsistency with the numpy function and had left some comments! Please request my review again once the changes are ready! Thanks a lot!

File: ivy/functional/frontends/numpy/indexing_routines/indexing_like_operations.py [hunk 3]
```diff
@@ -24,3 +24,8 @@ def diag(v, k=0):
 @to_ivy_arrays_and_back
 def take_along_axis(arr, indices, axis):
     return ivy.take_along_axis(arr, indices, axis)
+
+
+@to_ivy_arrays_and_back
+def tril_indices(n_rows, n_cols, k=0, /):
```
[the remark this answers is not in the document]
Comment (ipriyam26) [reply ivy/functional/frontends/numpy/indexing_routines/indexing_like_operations.py:30]: sure

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py (hunk 1 above)
Comment (ipriyam26) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py:142]: done

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py (hunk 2 above)
Comment (ipriyam26) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py:159]: I think this should be fine now

**Review by juliagsy — APPROVED** [review]
thanks for contributing!

**Pull request merged**